src/scripts/dbgap/dbgap_classes.py

```diff
@@ -49,55 +49,47 @@ def __init__(self, json_object, md5):
         sample_json = json_object[0]["attributes"]
         sample_id = json_object[0]['name']
         self._set_sample_attributes(sample_json, sample_id, md5)
-        self.set_telemetry_report_info()
 
     def _set_sample_attributes(self, sample_json, sample_id, md5):
-        self.project = sample_json["aggregation_project"]
-        self.location = sample_json["location"]
-        self.version = sample_json["version"]
-        self.md5 = md5
-        self.phs = str(sample_json["phs_id"])
-        self.data_type = sample_json["data_type"]
-        self.alias = sample_json["alias"]
-        self.file_type = self._get_file_extension(sample_json["aggregation_path"])
+        try:
+            self.project = sample_json["aggregation_project"]
+            self.location = sample_json["location"]
+            self.version = sample_json["version"]
+            self.md5 = md5
+            self.phs = str(sample_json["phs_id"])
+            self.data_type = sample_json["data_type"]
+            self.alias = sample_json["alias"]
+            self.aggregation_project = sample_json["aggregation_path"]
+        except KeyError as e:
+            raise ValueError(f"Missing required field: {e}")
+
+        self.file_type = self._get_file_extension(self.aggregation_project)
         self.data_file = f"{sample_id}.{self.file_type}"
-        self.set_telemetry_report_info()
+        self._dbgap_info = None
 
     @staticmethod
     def _get_file_extension(aggregation_path):
-        return aggregation_path.split(".")[-1]
-
-    def set_telemetry_report_info(self):
-        def is_bioproject_admin(xml_object):
-            return xml_object.attrib['bp_type'] == 'admin'
-        
-        def is_sample(xml_object):
-            return xml_object.attrib['submitted_sample_id'] == self.alias
-
-        root = ET.fromstring(call_telemetry_report(self.phs))
-        samples = [x.attrib for x in root.iter('Sample') if is_sample(x)]
-        bio_projects = [x.attrib['bp_id'] for x in root.iter('BioProject') if is_bioproject_admin(x)]
-
-        if not samples:
-            raise SampleNotRegisteredException('Sample not registered with Dbgap')
-        if not bio_projects:
-            raise StudyNotRegisteredException('Study not registered with Dbgap')
-        if len(samples) > 1:
-            raise Exception('Could not find specific sample in report')
-        
-        self.study = root[0].attrib
-        self.bio_project = bio_projects[0]
-        self.dbgap_sample_info = samples[0]
+        return os.path.splitext(aggregation_path)[1][1:]
 
+    @property
+    def dbgap_info(self):
+        if self._dbgap_info is None:
+            self._dbgap_info = DbgapTelemetryWrapper(phs_id=self.phs).get_sample_info(alias=self.alias)
+        return self._dbgap_info
```

Comment:
So it looks like you're calling `self._set_sample_attributes` in the init method, which is setting `self._dbgap_info` to None. But then in the `dbgap_info` property, you check if it's None, and set it to the `DbgapTelemetryWrapper` instance. I think you can probably skip all of that, and set `self.dbgap_info` in the init method directly. 

So something like: 

```
def __init__():
    self._set_sample_attributes(sample_json, sample_id, md5)
    self.dbgap_info = DbgapTelemetryWrapper(phs_id=self.phs).get_sample_info(alias=self.alias)
```

Reply:
Yea, I was trying to be clever and use lazy loading lol. Maybe it is just more confusing, I will switch it up